src/components/Board.js

```diff
@@ -1,33 +1,108 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import axios from 'axios';
-
 import './Board.css';
 import Card from './Card';
 import NewCardForm from './NewCardForm';
-import CARD_DATA from '../data/card-data.json';
+// import CARD_DATA from '../data/card-data.json';
 
 class Board extends Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
 
     this.state = {
       cards: [],
+      url: this.props.url,
+      boardName: this.props.boardName,
+      error: undefined
     };
   }
+  componentDidMount( ) {
+    const allCardsURL = this.state.url + 'boards/' + this.state.boardName + 'cards/';
 
-  render() {
-    return (
-      <div>
-        Board
-      </div>
-    )
+    axios.get(allCardsURL)
+    .then((response) => {
+
+      this.setState({
+        cards: response.data
+      })
+    })
+    .catch((error)=> {
+      const errorStr = `Got an error with status ${error.response.status} and message ${error.response.statusText}`
+
+      this.setState({
+        error: errorStr
+      })
+    });
+  }
+
+  deleteCard = (id) => {
+
+    const deleteURL = `${this.state.url}cards/${id}`
+    axios.delete(deleteURL)
+    .then((response) => {
+      const card = response.data.card;
+      alert(`Successfully deleted card with id ${card.id} and message ${card.text}.`)
+
+      const updatedCards = this.state.cards.filter((card) => {
+        return card.card.id !== id
+      })
+
+      this.setState({
+        cards: updatedCards
+      })
+    })
+    .catch((error) => {
+      const errorStr = `Got an error with status ${error.response.status} and message ${error.response.statusText}`
+      this.setState({
+        error: errorStr
+      })
+    })
   }
 
-}
+  addCard = (cardObj) => {
 
-Board.propTypes = {
+    let URL = `${this.state.url}boards/${this.state.boardName}cards`;
+
+    axios.post(URL, cardObj)
+    .then((response) => {
```

Comment:
I like that you've kept all the API interaction logic in one place - the callbacks are a little more complex, but I would say it makes the app as a whole much easier to comprehend. Whether or not you intended it, this is a great example of the container component pattern well-applied.